statsmodels/tsa/exponential_smoothing/_ets_smooth.pyx

```diff
@@ -0,0 +1,153 @@
+#!python
+#cython: wraparound=False, boundscheck=False, cdivision=True, annotate=True
+
+import numpy as np
+cimport numpy as np
+
+np.import_array()
+
+cpdef _initialize_ets_smooth(np.float_t [:] params, Py_ssize_t n):
+    """Extracts parameters and initializes states xhat"""
+    cdef np.float_t alpha, beta_star, gamma_star, phi, l0, b0
+    cdef np.float_t [:] s0
+    cdef np.float_t [:,:] xhat
+    cdef Py_ssize_t m
+
+    # get params
+    alpha, beta_star, gamma_star, phi = params[0:4]
+    m = len(params[6:])
+
+    # initialize states
+    # l = xhat[:,0], b = xhat[:,1], s = xhat[:,2:2+m]
+    # seasons are sorted such that xhat[:,2+m-1] contains s[-m]
+    xhat = np.zeros((n, 2 + m))
```

Comment:
Are these used in estimation?  If so, it might be worth considering passing chat in rather than reallocating it each iteration.

Reply:
Changed this. I added a `_loglike_internal` function that takes `yhat` and `xhat` as arguments, which is used during fitting, but Codacy complains about the replacement of `loglike` with `_loglike_internal`. Is there a better way to do this? I want to avoid duplicating the code from `statsmodels.base.model.fit`.